idna/codec.py

```diff
@@ -55,12 +57,13 @@ def _buffer_encode(self, data, errors, final):
             size += len(label)
 
         # Join with U+002E
-        result = '.'.join(result) + trailing_dot
+        result_str = '.'.join(result) + trailing_dot  # type: ignore
         size += len(trailing_dot)
-        return (result, size)
+        return result_str, size
```

Comment:
Again possibly not the correct moment to modify this within the current pull request, but I wonder if it's possible to simplify the logic around the returned string value and `size` variable.  In particular: is `size` always equal to `len(result_str)`?

Reply:
This whole file needs a lot of work, I mentioned it in initial PR message that I'm almost certain the `idna.codec` module is not functional. Would recommend solving all these problems in a different PR.